internal/controllers/gateway/gateway_utils.go

```diff
@@ -351,6 +352,13 @@ func getListenerStatus(
 			})
 		}
 
+		// consistent sort statuses to allow equality comparisons
+		sort.Slice(status.Conditions, func(i, j int) bool {
```

Comment:
BTW we might just create a func that will do that to keep our code DRY

Reply:
Is there a way to not use a dynamic scope anon function for `sort.Slice`? It looks like you have to, since the slice isn't an argument to the comparison function, but the comparison function needs access to the slice.

Reply:
maybe implement a `sortConditions` to sort slice of `metav1.Condition`?

Reply:
Eh, short enough that I'll take the wonky UX sort.Slice has to avoid Go type constraint stuff.